src/ray/gcs/gcs_client/service_based_gcs_client.h

```diff
@@ -59,7 +58,7 @@ class RAY_EXPORT ServiceBasedGcsClient : public GcsClient {
   /// Reconnect to GCS RPC server.
   void ReconnectGcsServer();
 
-  std::unique_ptr<RedisGcsClient> redis_gcs_client_;
+  std::shared_ptr<RedisClient> redis_client_;
```

Comment:
Why are we removing RedisGCSClient? We will support multiple backend including Redis right? We should use the interface here?

Reply:
em... I think the interface you mentioned is `GCSCLient` ? `RedisGcsClient ` is just a implements, BTW,  this pr is same as #9898 , but that pr is a little confusion, so I opened a new pr here.

Reply:
I think this change makes sense because previously `ServiceBasedGcsClient` depends on `RedisGcsClient`, which was a bad design pattern. Now we include a `RedisClient` in `ServiceBasedGcsClient` because `ServiceBasedGcsClient` requires pubsub functionality.

Reply:
Redis backend is not supported via RedisGCSClient, it's RedisStoreClient instead. 
`RedisGCSClient` was used for the original non-service-based GCS (raylet directly talks redis). It's deprecated now.